forte/processors/data_augment/algorithms/word_splitting_processor.py

```diff
@@ -0,0 +1,122 @@
+Augmented Text -> "I w ill be there so on."
+"""
+
+from math import ceil
+import random
+from typing import List, Iterable
+
+from forte.data.data_pack import DataPack
+from forte.data.multi_pack import MultiPack
+from forte.data.ontology import Annotation
+from forte.processors.data_augment import ReplacementDataAugmentProcessor
+from forte.utils.utils import get_class
+
+__all__ = ["RandomWordSplitDataAugmentProcessor"]
+
+
+class RandomWordSplitDataAugmentProcessor(ReplacementDataAugmentProcessor):
+    r"""
+    This class creates a processor to perform Random Word Splitting.
+    It randomly chooses n words in a sentence and splits each word at
+    a random position where n = alpha * input length.
+    """
+
+    def _augment(self, input_pack: MultiPack, aug_pack_names: List[str]):
+        augment_entry = get_class(self.configs["augment_entry"])
+
+        for pack_name in aug_pack_names:
+            data_pack: DataPack = input_pack.get_pack(pack_name)
+
+            annotations: List[Annotation] = []
+            indexes: List[int] = []
+            endings = []
+            annos: Iterable[Annotation] = data_pack.get(augment_entry)
+            for idx, anno in enumerate(annos):
+                annotations.append(anno)
+                indexes.append(idx)
+                endings.append(anno.end)
+            if len(annotations) > 0:
+                annotation_to_split = random.sample(
+                    [
+                        (anno, idx)
+                        for (anno, idx) in zip(annotations, indexes)
+                        if (anno.end - anno.begin) > 1
+                    ],
+                    ceil(self.configs["alpha"] * len(annotations)),
+                )
+                annotation_to_split = sorted(
+                    annotation_to_split, key=lambda x: x[1], reverse=True
+                )
+                for curr_anno in annotation_to_split:
+                    src_anno, src_idx = curr_anno
+                    splitting_position = random.randrange(
+                        1, (src_anno.end - src_anno.begin)
+                    )
+                    word_split = [
+                        src_anno.text[:splitting_position],
+                        src_anno.text[splitting_position:],
+                    ]
+                    if src_idx != 0:
+                        first_position = endings[src_idx - 1] + 1
+                        second_position = endings[src_idx]
+                        word_split[1] = " " + word_split[1]
+                    else:
+                        first_position = 0
+                        second_position = endings[0]
+                        word_split[1] = " " + word_split[1]
+
+                    self._insert(word_split[1], data_pack, second_position)
+                    self._delete(src_anno)
+                    self._insert(word_split[0], data_pack, first_position)
+
+    @classmethod
+    def default_configs(cls):
+        """
+        Returns:
```

Comment:
I just realize this is not included in the documentation, can you add it here: https://github.com/asyml/forte/blob/master/docs/code/data_aug.rst

Btw, can you also help add the typo op to the doc too: https://github.com/asyml/forte/blob/master/forte/processors/data_augment/algorithms/typo_replacement_op.py

Also, can you help check if the generated docstring is correct for this function? I think the indent and new lines can affect the outcome.

To check this (after you added them in the rst file above), perform `cd docs` and then `make html`, and then browse to the page to see if the outcome is expected.

I think this should be a good example: https://asyml-forte.readthedocs.io/en/latest/code/data.html#forte.data.batchers.ProcessingBatcher.default_configs